### ISSUE TITLE
Locking issue

### ISSUE THREAD

**Issue by J-shw** [itchannel/fordpass-ha#425]
My lock status seemed to have stopped working a few updates ago. It always shows 'unlocked' and if I try to lock the car using HA it throws this error:
```
File "/config/custom_components/fordpass/fordpass_new.py", line 264, in auth_step5
    if response.text is not None:
       ^^^^^^^^
UnboundLocalError: cannot access local variable 'response' where it is not associated with a value
```
Followed by this if I try again:
```
File "/config/custom_components/fordpass/fordpass_new.py", line 281, in auth
    raise Exception("Step 1 has reached error limit")
Exception: Step 1 has reached error limit
```

It started working when the 'double lock' feature was added. It would show my my car being locked, and would allow me to unlock/lock using HA. However it soon broke after more recent updates.

I have tried deleting the integration and redownloading it, logging out/in.

Also, thanks for all your hard work!

**Comment by danantonson** [comment]
Following, I have the same issue.   I removed my vehicle and re-added it through the integration and I no longer have the lock/unlock control.  It seems one of the updates broke/removed that control.  

Very much appreciate this integration.  Having automatically locking doors through HA has been amazing, I'd love to have it back.

**Comment by itchannel** [comment]
@danantonson Can you please provide me with the logs from HA, these should indicate why the sensor wasn't added. As the control is working for a lot of other users I expect it's something with your setup or vehicle so need to narrow down the cause. 

Can you provide the following information:
- Model and Year of vehicle
- Version of Fordpass integration
- Version of Home Assistant including install type e.g. HAOS, docker
- What is missing e.g. door entity

If you have time debug logging would also be beneficial 
https://github.com/itchannel/fordpass-ha/wiki/Obtaining-Integration-Logs

**Comment by danantonson** [comment]
Thanks for this @itchannel.  

Model and Year of vehicle:  **2018 Ford Escape**
Version of Fordpass integration **Version 1.59**
Version of Home Assistant including install type e.g. HAOS, docker  ** VMWare VM **
What is missing e.g. door entity ** lock.fordpass_doorlock **

I did the debugger; I need to sanitize the file, but the relevant lines I see are:

2023-10-28 22:06:50.269 WARNING (MainThread) [homeassistant.helpers.service] Referenced entities lock.fordpass_doorlock are missing or not currently available
2023-11-05 06:47:34.459 DEBUG (MainThread) [custom_components.fordpass.lock] Ford model doesn't support remote locking

Seeing that remote lock isn't supported prompted me to dig into my FordPass app (I never use it with this integration), and I noticed that I'm getting an error when I try to lock the doors through that. I wonder if that has something to do with it. I'll see if I can get on the line with Ford Tech. Support today and provide an update here.

![image](https://github.com/itchannel/fordpass-ha/assets/6033549/5aaeaccf-18c6-4684-aecd-d3e9828e308c)

**Comment by itchannel** [comment]
@danantonson Check your sync settings in your vehicle and make sure it's up to date to support the new Ford API. Also check your privacy settings in the vehicle and make sure connected is enabled.

**Comment by rkeberhardt** [comment]
Model and Year of vehicle: 2017 Ford Escape TI
Version of Fordpass integration Version 1.63
Version of Home Assistant including install type e.g. HAOS, Proxmox
What is missing e.g. door entity lock.fordpass_doorlock

I am also missing the doorlock.  Everything else seems to work, love the ability to automate preheating my car in the morning.  Unlike the previous commenter, I can still use the FordPass app to unlock and lock my car.  

I get the same error in logs.
2023-11-27 20:02:20.035 DEBUG (MainThread) [custom_components.fordpass.lock] Ford model doesn't support remote locking

I have no idea if this is a contributing factor, but two of the attributes of fordpass_doorStatus (hood and inner tailgate) always report "UNKNOWN".

**Comment by danantonson** [comment]
Has anyone made any progress on this or found any potential solutions? I updated my SYNC3 system, did a factory reset on that after the update which forced me to reauthenticate and setup FordPass itself.  I then reinstalled the HACS FordPass integration and, unfortunately, that didn't seem to make a difference. All I have is the ability to do the ignition; ability to lock still isn't there.  Any other ideas or suggestions?  

<img width="725" alt="Capture" src="https://github.com/itchannel/fordpass-ha/assets/6033549/e79266b0-1b68-42c2-8041-5b9f4dd47fa4">
